pkg/cluster/reconcile.go

```diff
@@ -151,6 +151,29 @@ func (c *Cluster) removeOneMember() error {
 }
 
 func (c *Cluster) removeDeadMember(toRemove *etcdutil.Member) error {
+	if c.cluster.Spec.SelfHosted != nil {
+		selectedNodes, err := c.selectMasterNodes()
+		if err != nil {
+			return err
+		}
+
+		// If we do not have enough master nodes, we should simply wait for the old node
+		// to be online again.
+		//
+		// Removing the etcd member will not help us to recover the cluster to the desired
+		// number of members, since we do not have enough master nodes for self hosted etcd.
+		//
+		// Instead, there is a large chance that the old node is taken offline for maintenance
+		// or experiencing temporary network partition. When it comes back, it will recover itself
+		// since we persist data for self hosted case.
+
+		if nodeNum := len(selectedNodes); nodeNum < c.cluster.Spec.Size {
```

Comment:
> The failed one will restart itself, right?

i do not understand. if you have 3 master nodes, and only two of them are running etcd. then it means you originally have 4 master nodes, and one of them is either dead or partitioned. it is reasonable to remove the member on that lost node and readd a member to the one without etcd.

can you provide me a full story of the case you have in mind? from the normal operation, to when it fails, and finally to what action the etcd operator is expected to take?